ddtrace/payload.py

```diff
@@ -0,0 +1,84 @@
+import logging
+
+from .encoding import get_encoder
+
+log = logging.getLogger(__name__)
+
+
+class Payload:
+    __slots__ = ['traces', 'size', 'encoder', 'max_payload_size', 'filters']
+
+    # Default max payload size of 5mb
+    # DEV: Trace agent limit is 10mb, cutoff at 5mb to ensure we don't hit 10mb
+    DEFAULT_MAX_PAYLOAD_SIZE = 5 * 1000000
+
+    def __init__(self, encoder=None, filters=None, max_payload_size=DEFAULT_MAX_PAYLOAD_SIZE):
+        self.filters = filters
+        self.max_payload_size = max_payload_size
+        self.encoder = encoder or get_encoder()
+        self.traces = []
+        self.size = 0
+
+    def filter(self, trace):
```

Comment:
Not sure if I like `filter`s being in the `Payload`. It's clean as is, but doesn't really feel like it's something the payload should be concerned about. To me it seems like a better fit in the writer. But again, since it's clean right now I'm fine either way.

Reply:
I agree, moved filter off of the payload and back into writer, that is where it should have been.